app/controllers/users_controller.rb

```diff
@@ -15,7 +15,11 @@ def create
   end
 
   def show
-    @user = current_user
+    if current_user
+      @user = current_user
+    else
+      render file: 'public/404'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.